kit/Kit.cpp

```diff
@@ -2365,11 +2365,40 @@ void Document::drainQueue()
         {
             std::vector<TileCombined> tileRequests = _queue->popWholeTileQueue();
 
-            // Put requests that include tiles in the visible area to the front to handle those first
-            std::partition(tileRequests.begin(), tileRequests.end(), [this](const TileCombined& req) {
-                return isTileRequestInsideVisibleArea(req); });
-            for (auto& tileCombined : tileRequests)
-                renderTiles(tileCombined);
+            const int consumedMicroS = std::chrono::duration_cast<std::chrono::microseconds>(
+                                           std::chrono::steady_clock::now() - now).count();
+            timeoutMicroS -= consumedMicroS;
+
+            if( timeoutMicroS >= consumedMicroS )
```

Comment:
This is curious to me. If we have a huge batch of tiles that will take 100's of milliseconds to render - some of them visible, some of them irrelevant - we will now go render all of them in if it didn't take very long to process the preceeding input events (?)

But we will not re-spin the poll to improve responsiveness or interactivity in this case - right ?